project/src/main/puzzle/tutorial/tutorial-hud.gd

```diff
@@ -4,46 +4,33 @@ extends Control
 UI items specific for puzzle tutorials.
 """
 
-export (NodePath) var puzzle_path: NodePath
-
-onready var _puzzle: Puzzle = get_node(puzzle_path)
-onready var _playfield: Playfield = _puzzle.get_playfield()
-onready var _piece_manager: PieceManager = _puzzle.get_piece_manager()
+# emitted when the HUD should be refreshed during initial setup or for a level change.
+signal refreshed
 
-# tracks what the player has done so far during this tutorial
-var _line_clears := 0
-var _box_clears := 0
-var _boxes_built := 0
-var _squish_moves := 0
-var _snack_stacks := 0
+export (NodePath) var puzzle_path: NodePath
 
-# tracks what the player did with the most recent piece
-var _did_line_clear := false
-var _did_box_clear := false
-var _did_build_box := false
-var _did_build_cake := false
-var _did_squish_move := false
+onready var puzzle: Puzzle = get_node(puzzle_path)
 
 func _ready() -> void:
 	visible = false
 	PuzzleScore.connect("game_prepared", self, "_on_PuzzleScore_game_prepared")
-	PuzzleScore.connect("game_started", self, "_on_PuzzleScore_game_started")
 	PuzzleScore.connect("after_level_changed", self, "_on_PuzzleScore_after_level_changed")
 	Level.connect("settings_changed", self, "_on_Level_settings_changed")
-	for skill_tally_item_obj in $SkillTallyItems/GridContainer.get_children():
-		var skill_tally_item: SkillTallyItem = skill_tally_item_obj
-		skill_tally_item.set_puzzle(_puzzle)
-	refresh()
+	replace_tutorial_module()
+
+
+func replace_tutorial_module() -> void:
```

Comment:
This warrants a comment.

Reply:
Added comment, ` Loads a new tutorial module corresponding to the current level`